lib/AST/ASTVerifier.cpp

```diff
@@ -1879,35 +1879,51 @@ class Verifier : public ASTWalker {
         Out << "MakeTemporarilyEscapableExpr type does not match subexpression";
         abort();
       }
-      
-      // Closure and opaque value should both be functions, with the closure
-      // noescape and the opaque value escapable but otherwise matching.
-      auto closureFnTy = E->getNonescapingClosureValue()->getType()
-        ->getAs<FunctionType>();
-      if (!closureFnTy) {
-        Out << "MakeTemporarilyEscapableExpr closure type is not a closure";
+
+      auto call = dyn_cast<CallExpr>(E->getSubExpr());
+      if (!call) {
+        Out << "MakeTemporarilyEscapableExpr subexpression is not a call\n";
         abort();
       }
```

Comment:
The subexpression needing to be a call doesn't seem like a necessary limitation. SILGen shouldn't care as long as the subexpression is well-formed. Would it suffice to verify that the subexpression type is an rvalue (and matches the type of the MakeTemporarilyEscapableExpr itself)?